src/libsync/abstractpropagateremotedeleteencrypted.cpp

```diff
@@ -36,11 +36,25 @@ QNetworkReply::NetworkError AbstractPropagateRemoteDeleteEncrypted::networkError
     return _networkError;
 }
 
+void AbstractPropagateRemoteDeleteEncrypted::setNetworkError(QNetworkReply::NetworkError err)
+{
+    if (_networkError == QNetworkReply::NetworkError::NoError) {
```

Comment:
I'm a bit on the fence regarding this... I'd expect such check to be done in the caller. A setter is expected to set and not swallow values as it sees fit.

Now one thing I wonder is the following: with your approach right now it means the first error will ultimately bubble up. With my stance it means the last error will ultimately bubble up.

Since we might want to declare more than one failure at removing a child, I wonder if that shouldn't be just a regular setter here and in PropagateRemoteDeleteEncryptedRootFolder log the individual failing items in a specific list and produce a specific error message based on that list: "Foo couldn't be removed because we failed at removing the following children: A, B, ...".

Applies similarly to setErrorString

Reply:
@er-vin But, that means if, for example, the first child fails, then the second one succeeds, and then the third one fails - we will fail because of a failure of the third (last) child? Is it proper logic? I've been thinking like this: the first failure - is the first and main reason to fail the entire removal of a root folder. Other children's failures would get displayed in logs.

The thing is when we get a failure like this, eventually, it will be displayed somewhere in the UI and we will see the error string and code for the last failed child. If this is what you want, I can remove conditions from within the setters.

Reply:
Right I see what you mean. OK, let's not overdo it but move the condition in the caller (when that makes sense).

Reply:
@er-vin In fact, I've added the setters because of conditions. The thing is, I had to put the condition into multiple places. The same condition that checks if the error was set already. In that case, there is no sense in having those setters anymore. I will remove them and bring back the duplicate conditions in the place of call. Is this fine?

Reply:
Either that or they need a better name. :-)

Setters which set only when they want reduce readability at the call site.

Reply:
@er-vin How about - "storeErrorString", "storNetworkError"?
Or, maybe "cacheErrorString", "cacheNetworkError"?

Also, can just name them:
- updateErrorString
- updateNetworkError

Do you like any of these?